# Introduction to TidyGenePlot

scRNA-seq analysis processed with the R package Seurat usually takes violin plots and feature plots to detect the expression pattern/levels of various genes. However, the violin plots generated by the function Vln() were often annoyed by
the element "Identify" and the legend on the plot;  as well as the feature plots annoyed by coordinates. To solve this problem, we developed the R package TidyGenePlot to generate tidy and clean violin plots and feature plots by removing those annoyed elements/factors, to give researchers a tidy and clean output.

# Dependencies
R package <br/>Seurat (version 3.1.2 or higher)<br/> patchwork (version 1.0.1 or higher) <br/> ggplot2 (version 3.5.1 or higher)

# Install TidyGenePlot
library(devtools) <br/>  devtools::install_github("LingzhangMeng/TidyGenePlot")

# User Guideline
Function 1: to generate tidy/clean violin plot(s) <br/> <br/> 
tidy.VlnPlot(seu_obj, features = feature.list, pt.size = pointsize, cols = colors, ncol = n)     <br/>
Explaination <br/> 
  seu_obj: seurat object created by R package Seurat <br/>
  feature.list: a vector contaiing genes <br/> 
  pointsize: pointsize used for drawing plot, number from 0 to 2 could be set, for example 0, 0.5, 1 or 2. <br/> 
  colors: could be set "NULL", which used the color panel from Seurat. Or could be customized by youself (color numbers should be equal or more than cluster numbers) <br/> 
  n: plot numbers arranged in each row, could be customized to numbers, such as 1, 2, 3, 4 5, 6...<br/> <br/> <br/> 


Function 2: to generate tidy/clean feature plot(s) <br/> <br/> 
tidy.FeaturePlot(seu_obj, features = feature.list, pt.size = pointsize, cols = colors, ncol = n)     <br/>
Explaination <br/> 
  seu_obj: seurat object created by R package Seurat <br/>
  feature.list: a vector contaiing genes <br/> 
  pointsize: pointsize used for drawing plot, number from 0 to 2 could be set, for example 0.5, 1 or 2. <br/> 
  colors: Do Not set it to "NULL", only 2 colors could be set, the first color will be used as backgroud, the second color will be used as gene expression level. I'd suggest set it to c("grey", "blue"), or set it to c("grey", "red") <br/> 
  n: plot numbers arranged in each row, could be customized to numbers, such as 1, 2, 3, 4 5, 6...<br/> <br/>
  # Examples and Comparison
  cb_palette <- c("#ed1299", "#09f9f5", "#246b93", "#cc8e12", "#d561dd", "#c93f00", "#ddd53e", "#4aef7b", 
                "#e86502", "#9ed84e", "#AB3282", "#CCC9E6", "#8249aa", "#99db27", "#DCC1DD", "#ff523f",
                "#ce2523", "#f7aa5d", "#cebb10", "#03827f", "#931635", "#373bbf", "#a1ce4c", "#ef3bb6", 
                "#d66551", "#1a918f", "#ff66fc", "#2927c4", "#7149af", "#57e559" ,"#8e3af4") <br/> <br/>
  VlnPlot(Control, features = c("Krt14", "Krt5", "Il1b", "Cd34", "Fxyd3", "Sfn"), pt.size = 0, ncol = 3) <br/>
  <img width="857" alt="1" src="https://github.com/user-attachments/assets/1ed5d17d-b990-4584-8c75-a27de92be5f7"> <br/> <br/>
  tidy.VlnPlot(Control, features = c("Krt14", "Krt5", "Il1b", "Cd34", "Fxyd3", "Sfn"), pt.size = 0, cols = cb_palette, ncol = 3) <br/>
  <img width="1235" alt="2" src="https://github.com/user-attachments/assets/7728885d-ef54-428f-bda0-77d369c8f325"> <br/> <br/>

  FeaturePlot(Control, features = c("Krt14", "Krt5", "Il1b", "Cd34", "Fxyd3", "Sfn"),  ncol = 3)  <br/>
  <img width="773" alt="3" src="https://github.com/user-attachments/assets/0c3ba7e6-8e70-411b-84a2-76399fd4ca1b"> <br/> <br/>
  
tidy.FeaturePlot(Control, features = c("Krt14", "Krt5", "Il1b", "Cd34", "Fxyd3", "Sfn"), cols = c("grey", "red"), ncol = 3) <br/>
<img width="770" alt="4" src="https://github.com/user-attachments/assets/8b498ee6-8699-4fc2-985e-468d11d44d57">











  
